tools/common/Application.cs

```diff
@@ -495,21 +498,49 @@ public void RunRegistrar ()
 
 			var ps = new ReaderParameters ();
 			ps.AssemblyResolver = resolver;
-			resolvedAssemblies.Add (ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
+			resolvedAssemblies.Add ("mscorlib", ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
+
+			var productAssembly = Driver.GetProductAssembly (this);
+			bool foundProductAssembly = false;
+			foreach (var asm in RootAssemblies) {
+				var rootName = Path.GetFileNameWithoutExtension (asm);
+				if (rootName == productAssembly)
+					foundProductAssembly = true;
+
+				Exception e = null;
+				AssemblyDefinition lastAssembly = null;
+				try {
+					lastAssembly = ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse (rootName), new ReaderParameters ());
+					if (resolvedAssemblies.TryGetValue (rootName, out var previousAssembly)) {
+						if (lastAssembly.MainModule.RuntimeVersion != previousAssembly.MainModule.RuntimeVersion) {
+							Driver.Log (2, "Attemping to load an assembly another time {0} (previous {1})", lastAssembly.FullName, previousAssembly.FullName);
+						}
+						continue;
+					}
+					
+					if (lastAssembly != null) {
+						resolvedAssemblies.Add (rootName, lastAssembly);
+						Driver.Log (3, "Loaded {0}", lastAssembly.MainModule.FileName);
+					}
+				} catch (Exception ex) {
+					e = ex;
+				}
 
-			var rootName = Path.GetFileNameWithoutExtension (RootAssembly);
-			if (rootName != Driver.GetProductAssembly (this))
-				throw ErrorHelper.CreateError (66, "Invalid build registrar assembly: {0}", RootAssembly);
+				if (lastAssembly == null)
```

Comment:
@rolfbjarne how do you want to handle erroring here? Crypto assemblies will be guaranteed to throw here. Should mmp stop or ignore and print out a message at log level 0?

Reply:
It can return null if it doesn't find the file, or throw a `ProductException(9)` if it can't load the assembly.